docs/source/pages/tutorials/commit_finder.rst

```diff
@@ -0,0 +1,162 @@
+   :alt: HTML report for ``arrow 0.15.0``, checks
+   :align: center
+
+In the second report for Arrow, we can see that Macaron has returned different results. Starting with the ``Target Information`` section we can see that the repository for this older artifact is not the same as the current one: ``https://github.com/crsmithdev/arrow`` instead of ``https://github.com/arrow-py/arrow``. In the checks section, we can see that two of the four checks that passed for the previous version, did not pass for this earlier version. Checks ``mcn_build_service_1`` and ``mcn_build_as_code_1`` failed, indicating that the older artifact did not have a GitHub Actions workflow setup to build or publish the package. In this way Macaron has demonstrated the usefulness of being able to analyze a repository at multiple stages, thereby allowing for a more accurate analysis when investigating artifacts that are, or use, outdated libraries.
+
+*****************************
+Run ``verify-policy`` command
+*****************************
+
+Another feature of Macaron is policy verification. This allows Macaron to report on whether an artifact meets the security requirements specified by the user. Policies are written using `Soufflé Datalog <https://souffle-lang.github.io/index.html>`_ , a language similar to SQL. Results collected by the ``analyze`` command can be checked via declarative queries in the created policy, which Macaron can then automatically check.
+
+The security requirement chosen for this tutorial reflects the difference between the two versions in the previous section. That is, we want to ensure that the artifact has a valid build service. If we refer back to :ref:`step <fig_arrow_0.15.0>` and :ref:`step <fig_arrow_1.3.0>`, we can see that the relevant check ID of the difference between the two versions is ``mcn_build_as_code_1``. To include this in a policy we create the following:
+
+.. code-block:: c++
+
+    #include "prelude.dl"
+
+    Policy("has-hosted-build", component_id, "Require a hosted build and publishing service.") :-
+        check_passed(component_id, "mcn_build_as_code_1").
+
+    apply_policy_to("has-hosted-build", component_id) :-
+        is_component(component_id, purl),
+        match("pkg:pypi/arrow.*", purl).
+
+The second part of the above policy, ``apply_policy_to``, applies the policy to software components found within Macaron's database based on the conditions within it. In this case, any software component whose PURL begins with ``pkg:pypi/arrow`` will be verified by the policy, thanks to Soufflé's `match constraint <https://souffle-lang.github.io/constraints>`_ that can be used for string matching. This will capture both versions of the Arrow library used in the previous section. To use the completed policy, we save it to an easily accessible location, such as the directory Macaron is in, with a name such as ``has-hosted-build.dl``. With the policy file created and saved, we can have Macaron run is as follows:
```

Comment:
```suggestion
The second part of the above policy, ``apply_policy_to``, applies the policy to software components found within Macaron's database based on the conditions within it. In this case, any software component whose PURL begins with ``pkg:pypi/arrow`` will be verified by the policy, thanks to Soufflé's `match constraint <https://souffle-lang.github.io/constraints>`_ that can be used for string matching. This will capture both versions of the Arrow library used in the previous section. To use the completed policy, we save it to an easily accessible location, such as the directory Macaron is in, with a name such as ``has-hosted-build.dl``. With the policy file created and saved, we can have Macaron run it as follows:
```

Reply:
See commit [4f11927](https://github.com/oracle/macaron/pull/606/commits/4f11927a8dd7d67b9aa1c61c31c980bba03f24fa)